CRM/Report/Form.php

```diff
@@ -4708,12 +4708,14 @@ public function getPermissionedFTQuery(&$query, $alias = NULL, $return = FALSE)
               {$query->_where}
                       AND {$query->_aliases['civicrm_contribution']}.financial_type_id IN (" . $contFTs . ")
                       AND {$query->_aliases['civicrm_line_item']}.id IS NULL
-              GROUP BY {$query->_aliases['civicrm_contribution']}.id";
+              GROUP BY {$query->_aliases['civicrm_contribution']}.id
+              UNION SELECT 0 as contribution_dummy_id
+    ";
     CRM_Core_DAO::executeQuery($sql);
     if (isset($temp)) {
       $query->_aliases['civicrm_line_item'] = $temp;
     }
-    $from = " INNER JOIN civicrm_contribution_temp temp ON {$query->_aliases['civicrm_contribution']}.id = temp.id ";
+    $from = " INNER JOIN civicrm_contribution_temp temp ON IF({$query->_aliases['civicrm_contribution']}.id IS NULL, 1, {$query->_aliases['civicrm_contribution']}.id = temp.id)  ";
```

Comment:
@JoeMurray Actually, we have approached this differently on instructions from the core team. We are now using the new selectWhereClause hook in an extension which will likely ship with core. Here is the code: https://github.com/JMAConsulting/biz.jmaconsulting.financialaclreport/blob/master/financialaclreport.php